components/strava/strava.app.mjs

```diff
@@ -0,0 +1,128 @@
+import { axios } from "@pipedream/platform";
+export default {
+  type: "app",
+  app: "strava",
+  methods: {
+    _getHeaders(headers = {}) {
+      return {
+        "Authorization": `Bearer ${this.$auth.oauth_access_token}`,
+        "Content-Type": "application/json",
+        "user-agent": "@PipedreamHQ/pipedream v0.1",
+        ...headers,
+      };
+    },
+    _getUrl(path) {
+      return `https://www.strava.com/api/v3${path}`;
+    },
+    async _makeRequest({
+      $,
+      path,
+      headers,
+      ...otherConfig
+    } = {}) {
+      const config = {
+        url: this._getUrl(path),
+        headers: this._getHeaders(headers),
+        ...otherConfig,
+      };
+      return axios($ ?? this, config);
+    },
+    async getAuthenticatedAthlete({
+      $,
+      ...otherConfig
+    } = {}) {
+      return await this._makeRequest({
+        $,
+        method: "GET",
+        path: "/athlete",
+        params: otherConfig,
+      });
+    },
+    async getActivity({
+      $,
+      activityId,
+      ...otherConfig
+    } = {}) {
+      return (
+        await this._makeRequest({
+          $,
+          method: "GET",
+          path: `/activities/${activityId}`,
+          params: otherConfig,
+        })
+      );
+    },
+    async getStats({
+      $,
+      athlete,
+      ...otherConfig
+    } = {}) {
+      return await this._makeRequest({
+        $,
+        method: "GET",
+        path: `/athletes/${athlete.id}/stats`,
+        params: otherConfig,
+      });
+    },
+    async createNewActivity({
+      $,
+      ...otherConfig
+    }) {
+      return await this._makeRequest({
+        $,
+        method: "POST",
+        path: "/activities",
+        data: otherConfig,
+      });
+    },
```

Comment:
```suggestion
    async createNewActivity(args = {}) {
      return this._makeRequest({
        method: "POST",
        path: "/activities",
        ...args,
      });
    },
```
Then if you want to use this post function from a component just do it like 
```javascript
await this.strava.createNewActivity({
  $,
  data: {...}
});
```

Reply:
done